src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -84,15 +85,22 @@ protected override ImmutableArray<ISymbol> GetSpecialIncompatibleMembers (Compil
 				incompatibleMembers.AddRange (methodInfoType.GetMembers ("MakeGenericMethod").OfType<IMethodSymbol> ());
 			}
 
+			var expressionCallType = compilation.GetTypeByMetadataName ("System.Linq.Expressions.Expression");
+			if (expressionCallType != null) {
+				incompatibleMembers.AddRange (expressionCallType.GetMembers ("Call").OfType<IMethodSymbol> ()
+					.Where (m => m.Parameters.Length == 4 && m.Parameters[0] is IParameterSymbol parameterSymbol
+					&& parameterSymbol.Type?.ContainingNamespace?.Name == "System" && parameterSymbol.Type?.Name == "Type"));
+			}
+
 			return incompatibleMembers.ToImmutable ();
 		}
 
 		protected override bool ReportSpecialIncompatibleMembersDiagnostic (OperationAnalysisContext operationContext, ImmutableArray<ISymbol> specialIncompatibleMembers, ISymbol member)
 		{
 			if (member is IMethodSymbol method && ImmutableArrayOperations.Contains (specialIncompatibleMembers, member, SymbolEqualityComparer.Default) &&
-				(method.Name == "MakeGenericMethod" || method.Name == "MakeGenericType")) {
-				// These two RUC-annotated APIs are intrinsically handled by the trimmer, which will not produce any
-				// RUC warning related to them. For unrecognized reflection patterns realted to generic type/method
+				(method.Name == "MakeGenericMethod" || method.Name == "MakeGenericType" || method.Name == "Call")) {
```

Comment:
Why do we need to check the method name if we are already comparing with `specialIncompatibleMembers`? I think we could get rid of the name check here. cc @tlakollo

Reply:
missed this duplicate check, thanks!